staking/types/messages.go

```diff
@@ -71,17 +79,14 @@ type Delegate struct {
 	Amount           *big.Int       `json:"amount" yaml:"amount"`
 }
 
-// Redelegate - type for reassigning delegation
-type Redelegate struct {
-	DelegatorAddress    common.Address `json:"delegator_address" yaml:"delegator_address"`
-	ValidatorSrcAddress common.Address `json:"validator_src_address" yaml:"validator_src_address"`
-	ValidatorDstAddress common.Address `json:"validator_dst_address" yaml:"validator_dst_address"`
-	Amount              *big.Int       `json:"amount" yaml:"amount"`
-}
-
 // Undelegate - type for removing delegation responsibility
 type Undelegate struct {
 	DelegatorAddress common.Address `json:"delegator_address" yaml:"delegator_address"`
 	ValidatorAddress common.Address `json:"validator_address" yaml:"validator_address"`
 	Amount           *big.Int       `json:"amount" yaml:"amount"`
 }
+
+// CollectRewards - type for collecting token rewards
+type CollectRewards struct {
```

Comment:
same question asked on the api design doc:

1. do we want to add amount of rewards to be collected here? or is it always going to be all rewards earned?

2. in our protocol code, can we easily look up the validator address given just the delegator address?

3. In Cosmos staking, do they have CollectRewards? It seemed they did not unless I missed it. How do they  enable collecting rewards?